# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/actors/BrowserElementParent.jsm
# Commit: bc3a104f8b20
# Full Hash: bc3a104f8b202e2501b6d0bee35ff60ab1d58733
# Author: Jonathan Watt <jwatt@jwatt.org>
# Date: 2019-12-21 09:45:32
# Regressor Bug: 1602410
# File Overlap Count: 3
# Description:
#   Bug 1602410. Make `window.print()` invoke printing via the parent process using a JSWindowActor. r=Gijs,kmag
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D56414
# ==============================================================================

diff -r 2e89a1d89e56 -r bc3a104f8b20 toolkit/actors/BrowserElementParent.jsm
--- a/toolkit/actors/BrowserElementParent.jsm	Sat Dec 21 00:46:20 2019 +0000
+++ b/toolkit/actors/BrowserElementParent.jsm	Sat Dec 21 00:22:42 2019 +0000
@@ -91,6 +91,17 @@
         }
         break;
       }
+      case "ContentRequestedPrint": {
+        // Note: this is not yet fully Fission compatible. That requires
+        // further work to change the PrintUtils API and the underlying code
+        // that it uses.
+        let bc = this.manager.browsingContext;
+        let topWin = bc.top.embedderElement.ownerGlobal;
+        if (topWin.PrintUtils) {
+          topWin.PrintUtils.printWindow(bc);
+        }
+        break;
+      }
     }
   }
 }
